Lesson2/src/Homework.java

```diff
@@ -0,0 +1,89 @@
+import java.util.Scanner;
+
+public class Homework {
+
+    public static void main(String[] args) {
+        System.out.println(sum(10, 20));
+        System.out.println(sum(Integer.MAX_VALUE, Integer.MAX_VALUE));
+        System.out.println(max(50, 49));
+        System.out.println(average(new int[]{1, 2, 3, 4, 5}));
+        System.out.println(max(new int[]{1, 2, 3, 4, 5, 100, 99}));
+        System.out.println(calculateHypotenuse());
+    }
+
+    /**
+     * 1. Method should return sum of "a" and "b"
+     * 2. (optional) if sum of a and b is more then int can store -- return -1;
+     **/
+    public static int sum(int a, int b) {
+        long t = (long)a+(long)b;
+        if ((long)(a+b) != t) return -1;
```

Comment:
when you write (long)(a+b) what happens on runtime - 
first, it sums a and b (as integers), then it casts the result to long. So if a = Integer.MAX_VALUE and b = Integer.MAX_VALUE, then
(long)(a+b) = -2 ,which is equal to just a + b

So you don't really need this cast to long here. Please let me know if you understood my comment =)

Reply:
ok i understand, this is redundant cast, delted.